node/pkg/processor/processor.go

```diff
@@ -234,31 +234,40 @@ func (p *Processor) Run(ctx context.Context) error {
 		return fmt.Errorf("workerFactor must be positive or zero")
 	}
 
+	var w sync.WaitGroup
+
+	w.Add(1)
+	go func(ctx context.Context) {
+		err := p.runHousekeeper(ctx)
+		if err != nil {
+			p.logger.Error("housekeeper failed", zap.Error(err))
+		}
+	}(ctx)
+
 	var ret error
 	if p.workerFactor == 0.0 {
-		ret = p.RunOne(ctx, 1)
+		w.Add(1)
+		ret = p.runWorker(ctx)
 	} else {
 		numWorkers := int(math.Ceil(float64(runtime.NumCPU()) * p.workerFactor))
 		p.logger.Info("processor configured to use workers", zap.Int("numWorkers", numWorkers), zap.Float64("workerFactor", p.workerFactor))
-
-		var w sync.WaitGroup
 		w.Add(numWorkers)
 
 		for workerId := 1; workerId <= numWorkers; workerId++ {
 			go func(ctx context.Context, workerId int) {
 				p.logger.Info("processor worker started", zap.Int("workerId", workerId))
-				err := p.RunOne(ctx, workerId)
+				err := p.runWorker(ctx)
 				if err != nil {
 					p.logger.Error("processor worker failed", zap.Int("workerId", workerId), zap.Error(err))
 				}
 				p.logger.Info("processor worker done", zap.Int("workerId", workerId))
 				w.Done()
 			}(ctx, workerId)
 		}
-
-		w.Wait()
 	}
 
+	w.Wait()
```

Comment:
Each worker should also use its own logger through `supervisor.Logger(ctx)` and `p.logger` should be removed. Otherwise it's hard to trace logs from multiple concurrent processes.

Reply:
`RunWithScissors` actually only catches panics, it doesn't do any re-starting. 
* I think panics in the processor should still lead to a Guardian panic and restarting the entire Guardian, so `RunWithScissors` should *not* be used here, although I'd be open to discussing that. 
* The routines should be restarted if they merely return an error. E.g. right now if the accountant returns an error the worker will just return that error. You can accomplish that by running the workers through the supervisor. The supervisor will restart runnables if they return an error or did not signal `supervisor.SignalDone` before returning.

Reply:
Ah I see, because of https://github.com/wormhole-foundation/wormhole/blob/02a33451b80552e47d133e0cb7c92c9d970bd21e/node/pkg/processor/processor.go#L272 the processor would return the error from any worker, which in turn would lead to the supervisor canceling all children and restarting the processor.